app/ide-desktop/lib/dashboard/src/providers/BackendProvider.tsx

```diff
@@ -70,14 +69,20 @@ export default function BackendProvider(props: BackendProviderProps) {
   )
 }
 
+function useBackendContext() {
+  const ctx = React.useContext(BackendContext)
+  if (ctx == null) throw new Error(`Backend not provided.`)
```

Comment:
Consider using tiny-invariant instead: https://www.npmjs.com/package/tiny-invariant

(we already added it in the project)